### PR TITLE
Add endpoint for removing a set of brokers forcefully

### LINKED ISSUE #16316: Add an endpoint for force remove brokers in gateway
Ref: https://github.com/camunda/zeebe/issues/16145#issuecomment-1945778387 

The API would be similar to cluster scaling. The input list contains the set of brokers that will remain. Every other broker will be force removed from the cluster.
```
POST actuator/cluster/brokers?force=true 
[
  <brokerId1>, <brokerId2>,..
]
```

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

This operation also changes the replicationFactor. Since it is hard-coded in the `BrokerInfo`, gateway now update it's in-memory state from the `ClusterTopology`. This makes the data in BrokerInfo useless. We should eventually remove this from BrokerInfo (related #14018).  In theory, it is possible to have different `replicationFactor` for each partition, especially when we use the fine-grained partition endpoint. To keep it simple, we always take the minimum replicationFactor available. 

## Related issues

closes #16316 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: zeebe/atomix/cluster/src/test/java/io/atomix/raft/ReconfigurationTest.java [hunk 1]
```diff
@@ -655,6 +655,24 @@ void shouldForceConfigureWhenRemovedMembersAreUnreachable() {
...
+    @Test
+    void shouldForceConfigureIfOnlyOneRemainingMember() {
+      // when
+      m2.shutdown().join();
+      m3.shutdown().join();
+      m4.shutdown().join();
+      m1.forceConfigure(Map.of(id1, Type.ACTIVE)).join();
+
+      // then
+      // leader must be one of m1 or m2
```
Comment (oleschoenburg) [inline zeebe/atomix/cluster/src/test/java/io/atomix/raft/ReconfigurationTest.java:667]:
```suggestion
      // leader must be m1
```

File: zeebe/broker-client/src/main/java/io/camunda/zeebe/broker/client/impl/BrokerTopologyManagerImpl.java [hunk 2]
```diff
@@ -212,10 +212,12 @@ public void onTopologyUpdated(final ClusterTopology clusterTopology) {
     }
     this.clusterTopology = clusterTopology;
     LOG.debug("Received new cluster topology with clusterSize {}", clusterTopology.clusterSize());
```
Comment (oleschoenburg) [inline zeebe/broker-client/src/main/java/io/camunda/zeebe/broker/client/impl/BrokerTopologyManagerImpl.java:214]: :upside_down_face: This log message is a both noisy _and_ not very informative. I'd suggest to check if it actually changed and only then log the entire topology or the actual change in cluster size, partition count and replication factor.

File: dist/src/main/java/io/camunda/zeebe/shared/management/ClusterEndpoint.java [hunk 3]
```diff
@@ -192,18 +193,19 @@ public ResponseEntity<?> scale(
...
-              .scaleMembers(
-                  new ScaleRequest(
-                      ids.stream()
-                          .map(String::valueOf)
-                          .map(MemberId::from)
-                          .collect(Collectors.toSet()),
-                      dryRun))
-              .join();
+          force
+              ? requestSender.forceScaleDown(scaleRequest).join()
```
Comment (oleschoenburg) [inline dist/src/main/java/io/camunda/zeebe/shared/management/ClusterEndpoint.java:207]: :question: Just do double check: `forceScaleDown` properly supports `dryRun`?

File: zeebe/qa/util/src/main/java/io/camunda/zeebe/qa/util/actuator/ClusterActuator.java [hunk 4]
```diff
@@ -122,6 +122,17 @@ PlannedOperationsResponse joinPartition(
   @Headers({"Content-Type: application/json", "Accept: application/json"})
   PlannedOperationsResponse scaleBrokers(@RequestBody List<Integer> ids, @Param boolean dryRun);
 
+  /**
+   * Scales the given brokers up or down and reassigns partitions to the new brokers.
+   *
+   * @param dryRun if true, changes are not applied but only simulated.
```
Comment (oleschoenburg) [inline zeebe/qa/util/src/main/java/io/camunda/zeebe/qa/util/actuator/ClusterActuator.java:128]: :wrench: Add the new `force` parameter.

File: zeebe/topology/src/main/java/io/camunda/zeebe/topology/state/ClusterTopology.java [hunk 5]
```diff
@@ -243,6 +243,20 @@ public int partitionCount() {
         members.values().stream().flatMap(m -> m.partitions().keySet().stream()).distinct().count();
   }
 
+  public Integer replicationFactor() {
```
Comment (oleschoenburg) [inline zeebe/topology/src/main/java/io/camunda/zeebe/topology/state/ClusterTopology.java:246]: :upside_down_face: I'd rename to `minReplicationFactor()` to make it clear that the replication factor can be different for each partition.

**Review by lenaschoenburg — APPROVED** [review]

File: dist/src/main/java/io/camunda/zeebe/shared/management/ClusterEndpoint.java (hunk 3 above)
Comment (deepthidevaki) [reply dist/src/main/java/io/camunda/zeebe/shared/management/ClusterEndpoint.java:207]: Yes. It follows the same approach as other operations.

File: zeebe/broker-client/src/main/java/io/camunda/zeebe/broker/client/impl/BrokerTopologyManagerImpl.java (hunk 2 above)
Comment (deepthidevaki) [reply zeebe/broker-client/src/main/java/io/camunda/zeebe/broker/client/impl/BrokerTopologyManagerImpl.java:214]: This should be already the case. This method is invoked only if local `ClusterTopology` has changed.

Comment (oleschoenburg) [reply zeebe/broker-client/src/main/java/io/camunda/zeebe/broker/client/impl/BrokerTopologyManagerImpl.java:214]: Ah okay :+1: 
During scaling we would still see repeated logs like "new cluster size 6" as operations get added and completed.

Comment (deepthidevaki) [reply zeebe/broker-client/src/main/java/io/camunda/zeebe/broker/client/impl/BrokerTopologyManagerImpl.java]: I changed it to log only if clusterSize or replicationFactor is changed. Instead, added a log in GatewayClusterTopologyService to log full topology if it has changed.

**Pull request merged**